### ISSUE TITLE
 error pipe:: Invalid data found when processing input

### ISSUE THREAD

**Issue by jshupeGitHub** [amiaopensource/vrecord#532]
I saw the following error
[decklink @ 0x561af5e99340] Device does not support selected audio input.
[decklink @ 0x561af5e99340] Could not set input configuration
DeckLink Mini Recorder: Input/output error
pipe:: Invalid data found when processing inputKB sq=    0B f=0/0   

 2020-04-22T16:00:58 -  [vrecord]Error: Running: /home/linuxbrew/.linuxbrew/opt/ffmpegdecklink/bin/ffmpeg-dl
Consider reporting this or asking for help at https://github.com/amiaopensource/vrecord/issues

I also see a
./vrecord: line 1492: defaults: command not found
--------------------------------
system ubuntu 18.04 64
mini recorder

error happened when attempting a record
the gui displayed Fine , the ability to edit also seemed fine

I had NO Signal  going to the card.
but I think that worked OK ( meaning no hookup / no actual sdi signal ) on a previous system
I will test with a signal sometime soon

in /home/linuxbrew/.linuxbrew/opt/ffmpegdecklink/bin
i have
20760440 Apr 20 15:16 ffmpeg-dl
20641688 Apr 20 15:16 ffplay-dl
20670520 Apr 20 15:16 ffprobe-dl
------------------------------------

thanks

jS

here is the whole shebang
scenesavers@ss-41:~/vrec$ ./vrecord -g
Gtk-Message: 15:59:53.570: Failed to load module "canberra-gtk-module"
Gtk-Message: 15:59:58.270: Failed to load module "canberra-gtk-module"
As auxiliary files directory was left blank, logs will be written to recording directory at /media/scenesavers/data_4/capture.
Variables set:
  DEVICE_INPUT_CHOICE = 0
  AVF_INPUT_CHOICE = 
  DECKLINK_INPUT_CHOICE = DeckLink Mini Recorder
  DIR = /media/scenesavers/data_4/capture
  LOGDIR = /media/scenesavers/data_4/capture
  CONTAINER_CHOICE = Matroska
  VIDEO_INPUT_CHOICE = Composite
  AUDIO_INPUT_CHOICE = Analog
  VIDEO_CODEC_CHOICE = FFV1 version 3
  FFV1_SLICE_CHOICE  = 24
  AUDIO_CODEC_CHOICE = 24-bit PCM
  VIDEO_BIT_DEPTH_CHOICE = 10 bit
  AUDIO_MAPPING_CHOICE = 2 Stereo Tracks (Channels 1 & 2 -> 1st Track Stereo, Channels 3 & 4 -> 2nd Track Stereo)
  TIMECODE_CHOICE = vitc
  STANDARD_CHOICE = NTSC
  QCTOOLSXML_CHOICE = No
  FRAMEMD5_CHOICE = No
  EMBED_LOGS_CHOICE = Yes
  PLAYBACKVIEW_CHOICE = Unfiltered
  PLAYBACKVIEW_CHOICE_PASS = Audio + Video
  DURATION = 
  TECHNICIAN = 
  PREFIX = 
  USER_SUFFIX = 
  NO_SUFFIX = true

Gtk-Message: 16:00:43.990: Failed to load module "canberra-gtk-module"
./vrecord: line 880: /usr/bin/mktemp: No such file or directory
Enter Identifier: jj
Enter the name of the person digitizing the tape or leave blank: shupert
Summary: FFV1 version 3/matroska yuv422p10 file from ntsc composite analog. Frame MD5s=No, QCTools XML=No, and Technician=shupert. Inputs recorded to /media/scenesavers/data_4/capture/jj.mkv and Auxiliary Files created in /media/scenesavers/data_4/capture
Hit enter to start recording

./vrecord: line 1492: defaults: command not found
Close the playback window to stop recording.
ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
  configuration: --prefix=/home/linuxbrew/.linuxbrew/Cellar/ffmpegdecklink/4.2.2_2 --disable-shared --enable-version3 --cc=gcc-7 --host-cflags= --host-ldflags= --enable-gpl --enable-libfreetype --enable-libmp3lame --enable-libopenjpeg --enable-libopus --enable-libsnappy --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libxvid --enable-libfontconfig --disable-libjack --disable-indev=jack --enable-libiec61883 --enable-nonfree --enable-decklink --extra-cflags=-I/home/linuxbrew/.linuxbrew/include --extra-ldflags=-L/home/linuxbrew/.linuxbrew/include
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[decklink @ 0x561af5e99340] Device does not support selected audio input.
[decklink @ 0x561af5e99340] Could not set input configuration
DeckLink Mini Recorder: Input/output error
pipe:: Invalid data found when processing inputKB sq=    0B f=0/0   

 2020-04-22T16:00:58 -  [vrecord]Error: Running: /home/linuxbrew/.linuxbrew/opt/ffmpegdecklink/bin/ffmpeg-dl
Consider reporting this or asking for help at https://github.com/amiaopensource/vrecord/issues

**Comment by dericed** [comment]
It looks like you selected Composite video and Analog audio, but doesn't the Decklink Mini only do SDI or HDMI?

**Comment by jshupeGitHub** [comment]
Yes - exactly correct {{ i have the red face }
I did have that wrong ( opps )
a edit error on my part

This is my second system , my second attempt at 
ubuntu 18 vrecord
1 diffrence is it uses gtkdialog - system wide - standard linux install
basically I have tried to depend "less" on linuxbrew

all previous remarks have been about the 1st system , 1st build
( it had the fail til i put in a symlink in usr/local/sbin to the linuxbrew gtkdialog )
The 1st one has made ~ 30 hours of capture .. i think successfully
-- but still in testing

this second one seems to still error

error
< console snip
scenesavers@ss-41:~/vrec$ ./vrecord -g
Gtk-Message: 16:52:34.257: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:52:36.187: Failed to load module "canberra-gtk-module"
Variables set:
  DEVICE_INPUT_CHOICE = 0
  AVF_INPUT_CHOICE = 
  DECKLINK_INPUT_CHOICE = DeckLink Mini Recorder
  DIR = /media/scenesavers/data_4/capture
  LOGDIR = /media/scenesavers/data_4/capture
  CONTAINER_CHOICE = Matroska
  VIDEO_INPUT_CHOICE = SDI
  AUDIO_INPUT_CHOICE = SDI Embedded Audio
  VIDEO_CODEC_CHOICE = FFV1 version 3
  FFV1_SLICE_CHOICE  = 24
  AUDIO_CODEC_CHOICE = 24-bit PCM
  VIDEO_BIT_DEPTH_CHOICE = 10 bit
  AUDIO_MAPPING_CHOICE = 2 Stereo Tracks (Channels 1 & 2 -> 1st Track Stereo, Channels 3 & 4 -> 2nd Track Stereo)
  TIMECODE_CHOICE = rp188vitc
  STANDARD_CHOICE = NTSC
  QCTOOLSXML_CHOICE = No
  FRAMEMD5_CHOICE = No
  EMBED_LOGS_CHOICE = No
  PLAYBACKVIEW_CHOICE = Unfiltered
  PLAYBACKVIEW_CHOICE_PASS = Unfiltered
  DURATION = 
  TECHNICIAN = 
  PREFIX = 
  USER_SUFFIX = 
  NO_SUFFIX = true

Gtk-Message: 16:53:29.359: Failed to load module "canberra-gtk-module"
./vrecord: line 880: /usr/bin/mktemp: No such file or directory
Enter Identifier: ll
Enter the name of the person digitizing the tape or leave blank: js
Summary: FFV1 version 3/matroska yuv422p10 file from ntsc sdi embedded. Frame MD5s=No, QCTools XML=No, and Technician=js. Inputs recorded to /media/scenesavers/data_4/capture/ll.mkv and Auxiliary Files created in /media/scenesavers/data_4/capture
Hit enter to start recording

./vrecord: line 1492: defaults: command not found
Close the playback window to stop recording.
ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
  configuration: --prefix=/home/linuxbrew/.linuxbrew/Cellar/ffmpegdecklink/4.2.2_2 --disable-shared --enable-version3 --cc=gcc-7 --host-cflags= --host-ldflags= --enable-gpl --enable-libfreetype --enable-libmp3lame --enable-libopenjpeg --enable-libopus --enable-libsnappy --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libxvid --enable-libfontconfig --disable-libjack --disable-indev=jack --enable-libiec61883 --enable-nonfree --enable-decklink --extra-cflags=-I/home/linuxbrew/.linuxbrew/include --extra-ldflags=-L/home/linuxbrew/.linuxbrew/include
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[decklink @ 0x55fa59432340] Found Decklink mode 720 x 486 with rate 29.97(i)
[decklink @ 0x55fa59432340] Frame received (#1) - No input signal detected - Frames dropped 1
Guessed Channel Layout for Input Stream #0.0 : 7.1
Input #0, decklink, from 'DeckLink Mini Recorder':
  Duration: N/A, start: 0.000000, bitrate: 236013 kb/s
    Stream #0:0: Audio: pcm_s32le, 48000 Hz, 7.1, s32, 12288 kb/s
    Stream #0:1: Video: v210 (V210 / 0x30313256), yuv422p10le(bottom first), 720x486, 223725 kb/s, 29.97 fps, 29.97 tbr, 1000k tbn, 1000k tbc
[Parsed_metadata_2 @ 0x55fa594518c0] Could not open : No such file or directory
[AVFilterGraph @ 0x55fa59454a80] Error initializing filter 'metadata' with args 'mode=print:file='
Error initializing complex filters.
No such file or directory
pipe:: Invalid data found when processing inputKB sq=    0B f=0/0   

 2020-04-22T16:53:44 -  [vrecord]Error: Running: /home/linuxbrew/.linuxbrew/opt/ffmpegdecklink/bin/ffmpeg-dl
Consider reporting this or asking for help at https://github.com/amiaopensource/vrecord/issues

</console snip end>

capture_options log
< start
computer_name: ss-41
computer_model_name: Intel(R) Xeon(R) CPU W3565 @ 3.20GHz
computer_cores: 8
user_name: scenesavers
operating_system_VERSION: #97-Ubuntu SMP Wed Apr 1 03:25:46 UTC 2020
vrecord version: 
datetime_start: 2020-04-22T16:53:43
FILE_PATH: /media/scenesavers/data_4/capture/ll.mkv
video_card_driver_version: 
video_card_name: DeckLink Mini Recorder
VIDEO_INPUT_CHOICE: SDI
AUDIO_INPUT_CHOICE: SDI Embedded Audio
VIDEO_BIT_DEPTH_CHOICE: 10 bit
AUDIO_MAPPING_CHOICE: 2 Stereo Tracks (Channels 1 & 2 -> 1st Track Stereo, Channels 3 & 4 -> 2nd Track Stereo)
TIMECODE_CHOICE: rp188vitc
STANDARD_CHOICE: NTSC
PLAYBACKVIEW_CHOICE: Unfiltered
CONTAINER_CHOICE: Matroska
VIDEO_CODEC_CHOICE: FFV1 version 3
FFV1_SLICE_CHOICE: 24
AUDIO_CODEC_CHOICE: 24-bit PCM
QCTOOLSXML_CHOICE: No
FRAMEMD5_CHOICE: No
EMBED_LOGS_CHOICE: No

TECHNICIAN: js
EXTRA_INPUT_OPTIONS: -timecode_format rp188vitc
FFmpeg command: /home/linuxbrew/.linuxbrew/opt/ffmpegdecklink/bin/ffmpeg-dl -nostdin -nostats -loglevel info -f decklink -draw_bars 0 -audio_input embedded -video_input sdi -format_code ntsc -channels 8 -audio_depth 32 -raw_format yuv422p10 -timecode_format rp188vitc -i DeckLink Mini Recorder -map_metadata 0:s:v:0 -color_primaries smpte170m -color_trc bt709 -colorspace smpte170m -color_range mpeg -metadata creation_time=now -c:v ffv1 -level 3 -g 1 -slices 24 -slicecrc 1 -c:a pcm_s24le -metadata:s:v:0 encoder=FFV1 version 3 -filter_complex [0:v:0]setsar=40/27,setdar=4/3,metadata=mode=print:file=; [0:a:0]pan=stereo| c0=c0 | c1=c1[stereo1];[0:a:0]pan=stereo| c0=c2 | c1=c3[stereo2] -map [stereo1] -map [stereo2] -f matroska /media/scenesavers/data_4/capture/ll.mkv -c copy -map 0 -f matroska -write_crc32 0 -live true -
FFmpeg command Error: 1

</ end log

**Comment by jshupeGitHub** [comment]
oh boy

ya know what I think it was 
I had timecode selected
but no actual signal hitting the card
select timecode none
and it stacks bytes
Complete name                            : /media/scenesavers/data_4/capture/3_41.mkv
Format                                   : Matroska
Format version                           : Version 4 / Version 2
File size                                : 6.54 MiB
Duration                                 : 13 s 246 ms
Overall bit rate                         : 4 142 kb/s
Encoded date                             : UTC 2020-04-23 20:44:02
Writing application                      : Lavf58.29.100
Writing library                          : Lavf58.29.100
ErrorDetectionType                       : Per level 1

Video
ID                                       : 1
Format                                   : FFV1
Format version                           : Version 3.4
Format settings, GOP                     : N=1
Codec ID                                 : V_MS/VFW/FOURCC / FFV1
Duration                                 : 13 s 246 ms
Width                                    : 720 pixels
Height                                   : 486 pixels
Display aspect ratio                     : 4:3
Frame rate mode                          : Constant
Frame rate                               : 29.970 (30000/1001) FPS
Standard                                 : NTSC
Color space                              : YUV
Chroma subsampling                       : 4:2:2
Bit depth                                : 10 bits
Scan type                                : Interlaced
Scan order                               : Bottom Field First
Compression mode                         : Lossless
Writing library                          : FFV1 version 3
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.601 NTSC
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.601
coder_type                               : Range Coder
MaxSlicesCount                           : 24
ErrorDetectionType                       : Per slice

Audio
ID                                       : 2
Format                                   : PCM
Format settings                          : Little / Signed
Codec ID                                 : A_PCM/INT/LIT
Duration                                 : 13 s 213 ms
Bit rate mode                            : Constant
Channel(s)                               : 2 channels
Sampling rate                            : 48.0 kHz
Bit depth                                : 24 bits
Writing library                          : Lavc58.54.100 pcm_s24le
Default                                  : Yes
Forced                                   : No

so .... no problem

but there is also this

it seems like yesterday when i did a ./vrecord -r

popping up in the linux dash was a icon for gtkdialog... the standard one.
it has a red & blue dot

NOW
i get a ffplay-dl to pop up in the linux dash

basically i was attempting to use a "system wide" install of gtkdialog not the linux brew

scenesavers@ss-41:~/vrec$ which gtkdialog
/usr/local/bin/gtkdialog

scenesavers@ss-41:~/vrec$ gtkdialog -v
gtkdialog version 0.8.3 release (C) 2003-2007 Laszlo Pere, 2011-2012 Thunor
Built with additional support for: none.

**Comment by dericed** [comment]
Should we close this or is there any documentation updates to consider?

**Comment by jshupeGitHub** [comment]
Hi Dave,

 

Not sure what this ticket was for.  Jim Shupert no longer works here.

 

Thank you,

Jim

 

 

 <http://www.theppsgroup.com/> 

Jim Bird

PPS Group

424 Scott Blvd.

Covington, KY 41011

Office  (859) 292-1150

Mobile (513) 708-4656

 <http://www.theppsgroup.com/> www.theppsgroup.com

**Comment by dericed** [comment]
will close, please re-open if needed
